pkg/rpcclient/rolemgmt/roles.go

```diff
@@ -0,0 +1,119 @@
+/*
+Package rolemgmt allows to work with the native RoleManagement contract via RPC.
+
+Safe methods are encapsulated into ContractReader structure while Contract provides
+various methods to perform the only RoleManagement state-changing call.
+*/
+package rolemgmt
+
+import (
+	"crypto/elliptic"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/native/nativenames"
+	"github.com/nspcc-dev/neo-go/pkg/core/native/noderoles"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/unwrap"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+)
+
+// Invoker is used by ContractReader to call various methods.
+type Invoker interface {
+	Call(contract util.Uint160, operation string, params ...interface{}) (*result.Invoke, error)
+}
+
+// Actor is used by Contract to create and send transactions.
+type Actor interface {
+	Invoker
+
+	MakeCall(contract util.Uint160, method string, params ...interface{}) (*transaction.Transaction, error)
+	MakeUnsignedCall(contract util.Uint160, method string, attrs []transaction.Attribute, params ...interface{}) (*transaction.Transaction, error)
+	SendCall(contract util.Uint160, method string, params ...interface{}) (util.Uint256, uint32, error)
+}
+
+// Hash stores the hash of the native RoleManagement contract.
+var Hash = state.CreateNativeContractHash(nativenames.Designation)
+
+const designateMethod = "designateAsRole"
```

Comment:
Hardly ever it will be replaced in future, but can we create a compat test checking that this const is compatible with the corresponding method name from native Designation manifest?

Reply:
I think it's covered by the client test, it won't work if you change this constant.